rerun_py/rerun_sdk/rerun/_rerun2/datatypes/angle_ext.py

```diff
@@ -11,3 +16,57 @@ def __init__(self: Any, rad: float | None = None, deg: float | None = None) -> N
             self.__attrs_init__(inner=deg, kind="degrees")  # pyright: ignore[reportGeneralTypeIssues]
         else:
             raise ValueError("Either `rad` or `deg` must be provided.")
+
+    @staticmethod
+    def native_to_pa_array_override(data: AngleArrayLike, data_type: pa.DataType) -> pa.Array:
+        from . import Angle
+
+        if isinstance(data, Angle) or isinstance(data, float):
+            data = [data]
+
+        types: list[int] = []
+        value_offsets: list[int] = []
+
+        num_nulls = 0
+        radians: list[float] = []
+        degrees: list[float] = []
+
+        null_type_idx = 0
+        radian_type_idx = 1
+        degree_type_idx = 2
+
+        for angle in data:
+            if angle is None:
+                value_offsets.append(num_nulls)
```

Comment:
Does the arrow spec (and some validation) require that these be increasing?

Reply:
I assumed, but let me try without!

Reply:
Box3d now has a test that hits a Rotation3D array with a `None` in it and I get a crash which is along side of what I expect if I don't do the `value_offset.append(num_nulls)` here
```
pyarrow.lib.ArrowInvalid: Buffer #2 too small in array of type dense_union<_null_markers: null=0, Quaternion: fixed_size_list<item: float not null>[4] not null=1, AxisAngle: struct<axis: fixed_size_list<item: float not null>[3] not null, angle: dense_union<_null_markers: null=0, Radians: float not null=1, Degrees: float not null=2> not null> not null=2> and length 5: expected at least 20 byte(s), got 16
```